From d441ebc447b68e72c4afbc1015abf1ce115cabfe Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 24 Jul 2020 12:00:06 +0200
Subject: [PATCH] libstdc++: do not pass const types to std::vector in autofill structured address component.

libstdc++ std::vector does not allow const value types:
/usr/include/c++/9/bits/stl_vector.h: In instantiation of ‘class std::vector<const std::__cxx11::basic_string<short unsigned int, base::string16_internals::string16_char_traits> >’:
../../components/autofill/core/browser/data_model/autofill_structured_address_component.cc:436:37:   required from here
/usr/include/c++/9/bits/stl_vector.h:398:66: error: static assertion failed: std::vector must have a non-const, non-volatile value_type

Bug: 957519
Change-Id: I0887c89a185749014a180cca8aec20f646bd16b7
---

diff --git a/components/autofill/core/browser/data_model/autofill_structured_address_component.cc b/components/autofill/core/browser/data_model/autofill_structured_address_component.cc
index 692fb5a..5953037 100644
--- a/components/autofill/core/browser/data_model/autofill_structured_address_component.cc
+++ b/components/autofill/core/browser/data_model/autofill_structured_address_component.cc
@@ -433,7 +433,7 @@
   result_pieces.reserve(10);
 
   // Store the inserted values to allow the used StringPieces to stay valid.
-  std::vector<const base::string16> inserted_values;
+  std::vector<base::string16> inserted_values;
   inserted_values.reserve(4);
 
   // Use a StringPiece rather than the string since this allows for getting
